### PR TITLE
[12.0] IMP sale_commission allowing to filter sale orders and invoices by agents

### PR DESCRIPTION
**Pull request by eLBati**

### CODE REVIEW

File: sale_commission/models/account_invoice.py [hunk 1]
```diff
@@ -19,6 +19,20 @@ def _compute_commission_total(self):
...
+    @api.multi
+    def _compute_agents(self):
+        for so in self:
+            so.agent_ids = [(6, 0, so.mapped("invoice_line_ids.agents.agent").ids)]
+
+    @api.model
+    def _search_agents(self, operator, value):
+        sol_agents = self.env["sale.order.line.agent"].search(
+            [("agent", operator, value)])
+        return [('id', 'in', sol_agents.object_id.order_id.ids)]
```
Comment (pedrobaeza) [inline sale_commission/models/account_invoice.py]: This is going to fail:
```suggestion
        return [('id', 'in', sol_agents.mapped("object_id.order_id").ids)]
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please add tests to cover the new code.

File: sale_commission/models/account_invoice.py (hunk 1 above)
Comment (eLBati) [reply sale_commission/models/account_invoice.py]: oops, fixed

**Comment by eLBati** [comment]
@pedrobaeza tests added

**Review by pedrobaeza — APPROVED** [review]

**Review by mrcast — APPROVED** [review]
Functional test ok!

**Comment by pedrobaeza** [comment]
Also interesting for v13.

**Review by marcelsavegnago — APPROVED** [review]

**Pull request merged**